### ISSUE TITLE
No longer able to build on device

### ISSUE THREAD

**Issue by amuresia** [aws-amplify/aws-sdk-ios#2242]
**Describe the bug**
I have returned from holiday and I am no able longer to build the app on my device. Building on simulator works without a problem. The source code had no changes according to git. I tried setting up a brand new dummy project with the instructions provided here https://github.com/aws-amplify/aws-sdk-ios#cocoapods but the error persists.

**To Reproduce**
1. Follow the tutorial to install dependencies via CocoaPods https://github.com/aws-amplify/aws-sdk-ios#cocoapods
2. Connect your iPhone via USB
3. Build your project
4. See error (below)

**Which AWS service(s) are affected?**
It seems like all?

**Expected behavior**
I would expect the project to be buildable and the application to operate like it used to

**Environment(please complete the following information):**
 - Package versions:
```
Installing AWSAutoScaling (2.12.6)
Installing AWSCloudWatch (2.12.6)
Installing AWSCognito (2.12.6)
Installing AWSCognitoIdentityProvider (2.12.6)
Installing AWSCognitoIdentityProviderASF (1.0.1)
Installing AWSCore (2.12.6)
Installing AWSDynamoDB (2.12.6)
Installing AWSEC2 (2.12.6)
Installing AWSElasticLoadBalancing (2.12.6)
Installing AWSIoT (2.12.6)
Installing AWSKinesis (2.12.6)
Installing AWSLambda (2.12.6)
Installing AWSMachineLearning (2.12.6)
Installing AWSMobileAnalytics (2.12.6)
Installing AWSS3 (2.12.6)
Installing AWSSES (2.12.6)
Installing AWSSNS (2.12.6)
Installing AWSSQS (2.12.6)
Installing AWSSimpleDB (2.12.6)
```
 - Dependency Manager: CocoaPods
 - Swift Version : 5.1

**Device Information (please complete the following information):**
 - Device: iPhone X
 - iOS Version: 13.3.1
 - Specific to simulators: No - it works perfectly fine on simulators

**Additional context**
```
dyld: Library not loaded: @rpath/AWSAutoScaling.framework/AWSAutoScaling
  Referenced from: /private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/newtest
  Reason: no suitable image found.  Did find:
	/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling: code signature invalid for '/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling'

	/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling: stat() failed with errno=25
	/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling: code signature invalid for '/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling'

	/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling: stat() failed with errno=1
	/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling: code signature invalid for '/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling'

	/private/var/containers/Bundle/Application/B4B3310C-D4BF-4071-895B-8EC89CFC052D/newtest.app/Frameworks/AWSAutoScaling.framework/AWSAutoScaling: stat() failed with errno=1
(lldb) 
```

You can see this happens to AWSAutoScaling, but in my main application it happens on AWSAuthCore

**Comment by lawmicha** [comment]
Hi @amuresia 

Could you try cleaning xcode directories like the derived data folders? https://stackoverflow.com/questions/32730312/reason-no-suitable-image-found

**Comment by amuresia** [comment]
Hi @lawmicha I have came across the same post last week. I tried again just now but still no joy.
I would also like to mention that I can build and deploy to mobile as long as I don't include Amplify dependencies in my Podfile.

**Comment by lawmicha** [comment]
thanks for trying, we'll see if we can reproduce it on our end. what do you mean by "Amplify" dependencies?

**Comment by amuresia** [comment]
With the following Podfile, my app crashes as soon as it's launched.
```
# Uncomment the next line to define a global platform for your project
 platform :ios, '13.3'

target 'testapp' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for testapp
  pod 'amplify-tools'
  pod 'Amplify'
  pod 'AWSPluginsCore'
  pod 'AmplifyPlugins'
 
end
```

If I remove the Amplify ones then it doesn't crash.

**Comment by lawmicha** [comment]
Just to clarify, the initial issue was with some of the existing AWS iOS SDK's like

```
Installing AWSAutoScaling (2.12.6)
Installing AWSCloudWatch (2.12.6)
Installing AWSCognito (2.12.6)
Installing AWSCognitoIdentityProvider (2.12.6)
Installing AWSCognitoIdentityProviderASF (1.0.1)
Installing AWSCore (2.12.6)
Installing AWSDynamoDB (2.12.6)
Installing AWSEC2 (2.12.6)
Installing AWSElasticLoadBalancing (2.12.6)
Installing AWSIoT (2.12.6)
Installing AWSKinesis (2.12.6)
Installing AWSLambda (2.12.6)
Installing AWSMachineLearning (2.12.6)
Installing AWSMobileAnalytics (2.12.6)
Installing AWSS3 (2.12.6)
Installing AWSSES (2.12.6)
Installing AWSSNS (2.12.6)
Installing AWSSQS (2.12.6)
Installing AWSSimpleDB (2.12.6)
```

but now your pod file is only the Amplify related pods from (https://github.com/aws-amplify/amplify-ios)

Just want to clarify, is this two separate problems or are do you have a set of pods that have ones from both AWS iOS SDk and Amplify? (the pod file you have above has no references to AWS iOS SDK pods)

**Comment by amuresia** [comment]
I've tried two approaches, both leading to the same problem.
1. Using the steps listed in https://github.com/aws-amplify/aws-sdk-ios#cocoapods
2. Using the steps listed in https://aws-amplify.github.io/docs/ios/datastore#using-cocoapods

The actual error message is exactly the same no matter which pods I use.
If you have a recommended Podfile that I could take into a blank project I'm happy to try that out.

**Comment by lawmicha** [comment]
Thanks for getting back to us @amuresia, if all pods are not working then I suspect the solution will lie in one of these SO answers https://stackoverflow.com/questions/32730312/reason-no-suitable-image-found .

- did you try `Xcode->Product->Clean + Xcode Restart`?
- or,  `Xcode -> Preferences -> Locations -> clicked the arrow behind the DerivedData path and removed the whole DerivedData folder in the Finder.` ?

I would start off fresh with just a single dependency that you may be interested in using. For example, https://aws-amplify.github.io/docs/sdk/ios/storage will result in a pod file with this

```
pod 'AWSS3', '~> 2.12.0'   # For file transfers
```
or following the readme in this sample will provision S3, install the pod dependencies, and get the app running: https://github.com/awslabs/aws-sdk-ios-samples/tree/master/S3TransferUtility-Sample/Swift

**Comment by SaurabhPrajapati** [comment]
Any Update on this? I'm also facing same issue on iOS 13.3.1

**Comment by palpatim** [comment]
@amuresia, @SaurabhPrajapati Thanks for reporting. I've been able to reproduce this locally and am investigating now.

**Comment by palpatim** [comment]
This appears to be a problem when installing any app...
- ...to a physical device running iOS 13.3.1
- ...that uses any embedded framework (not just AWS)
- ...and which has been code signed by a free Apple developer account

See the [Apple Forums thread](https://forums.developer.apple.com/message/404653)

Workarounds:

Because this appears to be an Apple bug, and is not isolated to AWS, the only good workarounds are:

- Install to a simulator rather than a physical device
- Use a paid account, or ensure that the enterprise/business account profile you're using does not say "(Personal Account)" in the title

See some discussions on this thread: https://github.com/flutter/flutter/issues/49504#issuecomment-581554697

I do verify that my sample app that includes AWS SDK pods builds fine using my personal paid developer account.

We'll keep an eye on this--hopefully this will be addressed in an upcoming release of iOS, or the Xcode toolchain. I haven't tested building using Xcode 11.4 beta, but that's next on my list to try.

**Comment by palpatim** [comment]
Marking this as "Pending" to indicate that we're waiting on a fix from Apple.

**Comment by palpatim** [comment]
Unfortunately I can't test on 11.4 beta because I haven't yet upgraded to Catalina. Once I get upgraded I'll test (assuming the issue isn't resolved via other means by then)

**Comment by ChrisInspect** [comment]
Same Issue here with AWSCore in the errorDescription since my iPad is updated to iOS 13.3.1

**Comment by lewystanton** [comment]
Can confirm that switching from capacitor to cordova works. While not ideal, I can now at least build apps on a real device again. The steps I took were;

```
npm uninstall --save @capacitor/cli @capacitor/core
rm -rf node_modules
rm -rf ios  (this one fixes inability to add ios platform)
npm install
npm i -g cordova
npm i -g ios-deploy
ionic cordova platform add ios
ionic cordova run ios --prod
```

There's probably a more efficient way to do this but it seems to work.

**Comment by amuresia** [comment]
I can confirm that after upgrading my account to a paid plan the issue disappeared.

**Comment by palpatim** [comment]
This appears to have been fixed in iOS 13.4.
